autoPyTorch/api/base_task.py

```diff
@@ -1079,73 +1135,223 @@ def refit(
+                will be per job. This memory limit also applies to the ensemble
+                creation process.
+            eval_metric (str): name of the metric that is used to
+                evaluate a pipeline.
+            all_supported_metrics (bool), (default=True): if True, all
+                metrics supporting current task will be calculated
+                for each pipeline and results will be available via cv_results
+            budget_type (Optional[str]):
+                Type of budget to be used when fitting the pipeline.
+                Either 'epochs' or 'runtime'. If not provided, uses
+                the default in the pipeline config ('epochs')
+            include_components (Optional[Dict]): If None, all possible components are used.
+                Otherwise specifies set of components to use.
+            exclude_components (Optional[Dict]): If None, all possible components are used.
+                Otherwise specifies set of components not to use. Incompatible with include
+                components
+            search_space_updates(Optional[HyperparameterSearchSpaceUpdates]): updates to be made
+                to the hyperparameter search space of the pipeline
+            budget (Optional[float]):
+                Budget to fit a single run of the pipeline. If not
+                provided, uses the default in the pipeline config
+            pipeline_options (Optional[Dict]):
+                Valid config options include "device",
+                "torch_num_threads", "early_stopping", "use_tensorboard_logger",
+                "metrics_during_training"
+            disable_file_output (Optional[Union[bool, List]]):
+                By default, the model, it's predictions and other metadata is stored on disk
+                for each finished configuration. This argument allows the user to skip
+                saving certain file type, for example the model, from being written to disk.
+            configuration: (Optional[Configuration])
                 configuration to fit the pipeline with. If None,
-                uses default
+                uses default.
 
         Returns:
             (BasePipeline): fitted pipeline
+            (RunInfo): Run information
+            (RunValue): Result of fitting the pipeline
+            (BaseDataset): Dataset created from the given tensors
         """
-        if self.dataset_name is None:
-            self.dataset_name = str(uuid.uuid1(clock_seq=os.getpid()))
 
-        if self._logger is None:
-            self._logger = self._get_logger(self.dataset_name)
+        resampling_strategy = resampling_strategy if resampling_strategy is not None else self.resampling_strategy
+        resampling_strategy_args = resampling_strategy_args if resampling_strategy_args is not None else \
+            self.resampling_strategy_args
+
+        dataset = self._create_dataset(X_train=X_train,
```

Comment:
If we enhance the create_dataset, we can call fit_pipeline multiple times without the need for a new dataset each time. Plus the training split will not change. Also, the name then probably should be `get_dataset`